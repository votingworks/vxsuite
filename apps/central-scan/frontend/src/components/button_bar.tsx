import styled from 'styled-components';

interface Props {
  dark?: boolean;
  naturalOrder?: boolean;
  secondary?: boolean;
  separatePrimaryButton?: boolean;
  centerOnlyChild?: boolean;
}

export const ButtonBar = styled('nav')<Props>`
  display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  order: ${({ secondary }) => (secondary ? '-1' : undefined)};
  border-bottom: 1px solid rgb(169, 169, 169);
  background: ${({ dark = true }) =>
    dark ? '#455a64' : 'rgba(0, 0, 0, 0.05)'};
  padding: 0.5rem;

  & > * {
    white-space: nowrap;
    flex: 1;
    margin: 0.25rem;

    @media (width >= 480px) {
      flex: ${({ separatePrimaryButton }) =>
        separatePrimaryButton ? '0' : undefined};
    }
  }

  & > *:first-child {
    order: ${({ naturalOrder = false }) => (naturalOrder ? undefined : '2')};

    @media (width >= 480px) {
      margin-right: ${({ naturalOrder = false, separatePrimaryButton }) =>
        separatePrimaryButton && naturalOrder ? 'auto' : undefined};
      margin-left: ${({ naturalOrder = false, separatePrimaryButton }) =>
        separatePrimaryButton && !naturalOrder ? 'auto' : undefined};
    }
  }

  & > *:only-child {
    @media (width >= 480px) {
      flex: ${({ centerOnlyChild = true }) => (centerOnlyChild ? 0 : 1)};
      margin: ${({ centerOnlyChild = true }) =>
        centerOnlyChild ? 'auto' : undefined};
      min-width: 33.333%;
    }
  }

  @media print {
    display: none;
  }
`;
