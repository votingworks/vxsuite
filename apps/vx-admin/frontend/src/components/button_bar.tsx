import styled from 'styled-components';

interface Props {
  dark?: boolean;
  primaryRight?: boolean;
  padded?: boolean;
}

export const ButtonBar = styled('nav')<Props>`
  display: flex;
  flex-wrap: wrap-reverse;
  align-items: center;
  justify-content: space-between;
  background: ${({ dark = false }) =>
    dark ? '#333333' : 'rgba(0, 0, 0, 0.05)'};
  padding: ${({ padded = false }) => (padded ? '0.25rem' : undefined)};

  & > * {
    flex-grow: 1;
    margin: 0.25rem;
  }

  & > *:first-child {
    order: ${({ primaryRight = false }) => (primaryRight ? 2 : undefined)};
  }
`;
