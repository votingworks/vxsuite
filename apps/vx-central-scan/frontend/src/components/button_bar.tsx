import styled from 'styled-components';

interface Props {
  dark?: boolean;
  naturalOrder?: boolean;
  secondary?: boolean;
  separatePrimaryButton?: boolean;
  centerOnlyChild?: boolean;
}

export const ButtonBar = styled('nav')<Props>`
  display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  order: ${({ secondary }) => (secondary ? '-1' : undefined)};
  border-bottom: 1px solid rgb(169, 169, 169);
  background: ${({ dark = true }) =>
    dark ? '#455a64' : 'rgba(0, 0, 0, 0.05)'};
  padding: 0.5rem;

  & > *:first-child {
    order: ${({ naturalOrder = false }) => (naturalOrder ? undefined : '2')};
    @media (min-width: 480px) {
      margin-right: ${({ naturalOrder = false, separatePrimaryButton }) =>
        separatePrimaryButton && naturalOrder ? 'auto' : undefined};
      margin-left: ${({ naturalOrder = false, separatePrimaryButton }) =>
        separatePrimaryButton && !naturalOrder ? 'auto' : undefined};
    }
  }

  & > * {
    white-space: nowrap;
    flex: 1;
    margin: 0.25rem;
    @media (min-width: 480px) {
      flex: ${({ separatePrimaryButton }) =>
        separatePrimaryButton ? '0' : undefined};
    }
  }
  & > *:only-child {
    @media (min-width: 480px) {
      flex: ${({ centerOnlyChild = true }) => (centerOnlyChild ? 0 : 1)};
      margin: ${({ centerOnlyChild = true }) =>
        centerOnlyChild ? 'auto' : undefined};
      min-width: 33.333%;
    }
  }
  @media print {
    display: none;
  }
`;
