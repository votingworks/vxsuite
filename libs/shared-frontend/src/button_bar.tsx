import styled from 'styled-components';

export const ButtonBar = styled('nav')`
  display: flex;
  flex-wrap: wrap-reverse;
  align-items: center;
  justify-content: space-between;
  border-bottom: 1px solid rgb(169, 169, 169);
  background: rgba(0, 0, 0, 0.05);
  padding: 0.25rem;

  & > *:first-child {
    order: 2;
    min-width: 50%;
  }

  & > * {
    flex-grow: 1;
    margin: 0.25rem;
  }
  & > *:only-child {
    @media (min-width: 480px) {
      flex-grow: initial;
      margin: auto;
      min-width: 33.333%;
    }
  }
`;
