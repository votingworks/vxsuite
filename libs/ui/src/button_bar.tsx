import styled from 'styled-components';

export const ButtonBar = styled('div')`
  display: flex;
  flex-wrap: wrap-reverse;
  align-items: center;
  justify-content: center;
  border-top: ${(p) => p.theme.sizes.bordersRem.hairline}rem solid
    ${(p) => p.theme.colors.foreground};
  padding: ${(p) => p.theme.sizes.minTouchAreaSeparationPx}px;
  gap: ${(p) => p.theme.sizes.minTouchAreaSeparationPx}px;

  & > * {
    flex-grow: 1;
  }

  & > *:first-child {
    order: 2;
    min-width: 40%;
  }

  & > *:only-child {
    @media (min-width: 480px) {
      flex-grow: initial;
      margin: auto;
      min-width: 33.333%;
    }
  }
`;
