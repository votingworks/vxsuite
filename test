#!/usr/bin/env bun

import { readFile } from 'fs/promises';
import { join } from 'path';

const { $ } = Bun;
export {};

const workspacePackageJson = await readFile(
  join(__dirname, 'package.json'),
  'utf8'
);
const { workspaces } = JSON.parse(workspacePackageJson);
const WIDTH = 100;

const results: { pkgPath: string; status: 'success' | 'failure' }[] = [];
let totalPassed = 0;
let totalFailed = 0;

for (const pkgPath of workspaces) {
  const tests = box(
    (async function* () {
      yield `→ bun run test --cwd ${pkgPath}`;
      const bunTest = $`bun run test --cwd ${pkgPath}`.quiet();

      const bunTestResult = await bunTest;

      yield* bunTestResult.stderr.toString('utf-8').split('\n');
      yield `${bunTestResult.exitCode === 0 ? '✅' : '❌'} ${pkgPath}`;

      if (bunTestResult.exitCode === 0) {
        results.push({ pkgPath, status: 'success' });
      } else {
        results.push({ pkgPath, status: 'failure' });
      }
    })()
  );

  let passed = 0;
  let failed = 0;

  for await (const line of tests) {
    const passedMatch = line.match(/(\d+) pass/);
    const failedMatch = line.match(/(\d+) fail/);

    if (passedMatch) {
      passed = parseInt(passedMatch[1], 10);
    }

    if (failedMatch) {
      failed = parseInt(failedMatch[1], 10);
    }

    console.log(line);
  }

  totalPassed += passed;
  totalFailed += failed;
}

console.log(
  `\n${totalPassed + totalFailed} tests: ${totalPassed} passed, ${totalFailed} failed`
);

for (const { pkgPath, status } of results) {
  console.log(`${status === 'success' ? '✅' : '❌'} ${pkgPath}`);
}

async function* box(lines: AsyncIterable<string>): AsyncIterable<string> {
  yield* lines;
}
