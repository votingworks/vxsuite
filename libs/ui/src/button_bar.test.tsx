import React from 'react';
import { render } from '../test/react_testing_library';

import { ButtonBar } from './button_bar';

test('renders ButtonBar', () => {
  const { container } = render(<ButtonBar>foo</ButtonBar>);
  expect(container.firstChild).toMatchSnapshot();
});
