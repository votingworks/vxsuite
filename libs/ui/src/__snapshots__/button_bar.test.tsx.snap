// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ButtonBar 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-top: 0.06rem solid #222222;
  padding: 7.2px;
  gap: 7.2px;
}

.c0 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 > *:first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  min-width: 40%;
}

@media print {

}

@media print {

}

@media print {

}

@media (min-width:480px) {
  .c0 > *:only-child {
    -webkit-box-flex: initial;
    -webkit-flex-grow: initial;
    -ms-flex-positive: initial;
    flex-grow: initial;
    margin: auto;
    min-width: 33.333%;
  }
}

<div
  class="c0"
>
  foo
</div>
`;
